source/Ship.cpp

```diff
@@ -3729,31 +3729,37 @@ void Ship::DoGeneration()
 		// 4. Shields of carried fighters
 		// 5. Transfer of excess energy and fuel to carried fighters.
 
-		const double hullAvailable = attributes.Get("hull repair rate")
+		const double hullAvailable = (attributes.Get("hull repair rate")
+			+ (hullDelay ? 0 : attributes.Get("delayed hull repair rate")))
 			* (1. + attributes.Get("hull repair multiplier"));
-		const double hullEnergy = (attributes.Get("hull energy")
+		const double hullEnergy = ((attributes.Get("hull energy")
+			+ (hullDelay ? 0 : attributes.Get("delayed hull energy")))
 			* (1. + attributes.Get("hull energy multiplier"))) / hullAvailable;
```

Comment:
```suggestion
		const double hullEnergy = (attributes.Get("hull energy")
			+ (hullDelay ? 0 : attributes.Get("delayed hull energy")))
			* (1. + attributes.Get("hull energy multiplier")) / hullAvailable;
```
We don't need this as the equation is processed left-to-right; same for other lines starting with double `(`.

Reply:
https://github.com/endless-sky/endless-sky/pull/9771/commits/cd07cf81bf96cde8b25f1dacd96fb494674e0c3b isn't really what I meant. In the above example, there are completely redundant brackets enclosing everything before a division operator, while your commit affects the calculations (multiplication takes precedence over addition).